src/buggy_code.py

```
@@ -0,0 +1,56 @@
+        return l
+
+def helper_function(x,y):
+    '''
+    Helper function to process data
+    '''
+    z = x+y
+    print(f"Result: {z}")
+    return z
+
+def process_complex_data(input_data, transformation_factor, processing_mode, additional_parameters, debug_mode=False):
+    if debug_mode == True:
```

Ruff

Avoid equality comparisons to `True`; use `if debug_mode:` for truth checks

Replace with `debug_mode`

(E712)